### PR TITLE
Fix travis-ci API permission error

### PR DESCRIPTION
**Pull request by isuruf**
Travis sometimes knows about github repos, but doesn't know that the bot
has admin access to it. (Maybe because travis accounts of cf admins are
synced, travis knows about the repo)

In that case, sync account and wait till travis knows that the bot has
admin access

### CODE REVIEW

File: conda_smithy/ci_register.py [hunk 1]
```diff
@@ -214,13 +214,19 @@ def travis_wait_until_synced(ignore=False):
     return content
 
 
-def travis_get_repo_info(user, project, show_error=False):
+def travis_get_repo(user, project, show_error=False, check_admin_access=True):
```
Comment (pelson) [inline conda_smithy/ci_register.py]: Why add a boolean flag only to never use it in any of the code?

File: conda_smithy/ci_register.py [hunk 2]
```diff
@@ -214,13 +214,19 @@ def travis_wait_until_synced(ignore=False):
...
     headers = travis_headers()
     url = '{}/repo/{user}%2F{project}'.format(travis_endpoint, user=user, project=project)
     response = requests.get(url, headers=headers)
     try:
         response.raise_for_status()
         content = response.json()
+        if check_admin_access:
```
Comment (pelson) [inline conda_smithy/ci_register.py]: I'd take all the return statements out of the ``try..except`` block at this point - the exception is getting too far away from the place where the exception may be getting raised.

File: conda_smithy/ci_register.py [hunk 3]
```diff
@@ -214,13 +214,19 @@ def travis_wait_until_synced(ignore=False):
     return content
 
 
-def travis_get_repo_info(user, project, show_error=False):
```
Comment (pelson) [inline conda_smithy/ci_register.py]: Why change the function name in this PR?
Personally, I feel like the "info" part captures the fact that there is no such thing as a "repo" object...

**Comment by scopatz** [comment]
Thanks all!

**Pull request merged**